core/sdk/src/extn/client/extn_client.rs

```diff
@@ -121,17 +121,33 @@ impl ExtnClient {
     ///
     /// Also starts the thread in the processor to accept incoming requests.
     pub fn add_request_processor(&mut self, mut processor: impl ExtnRequestProcessor) {
+        let contracts = if let Some(multiple_contracts) = processor.fulfills_mutiple() {
+            multiple_contracts
+        } else {
+            vec![processor.contract()]
+        };
+        let mut atleast_one_contract_satisfied = false;
```

Comment:
Do you need this variable? Is it not the same as `!contracts_supported.is_empty()`

Reply:
Actually they depend on whether this contract is enabled on the extension manifest. This uses `check_contract_fulfillment()` method to see if a given extension has the entry in manifest. If not it will be ignored and not setup.

Reply:
Added your patch